instrumentation/aws_lambda/lib/opentelemetry/instrumentation/aws_lambda/handler.rb

```diff
@@ -47,6 +48,9 @@ def call_wrapped(event:, context:)
               kind: span_kind
             )
 
+            trace_ctx = OpenTelemetry::Trace.context_with_span(span)
+            trace_token = OpenTelemetry::Context.attach(trace_ctx)
```

Comment:
@xuan-cao-swi I'm curious whether you tried using `tracer.in_span` instead of `tracer.start_span` and having to explicitly manage the context. 

Since you're already using `OpenTelemetry::Context.with_current(parent_context)` in line 43 above to propagate any incoming context into the current one, `in_span` should be able to continue that and as a benefit record exception too. Some examples in exising "[server](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/542fa91b796c4abb1b855b91c1af1d67894a2798/instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb)" or "[consumer](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/542fa91b796c4abb1b855b91c1af1d67894a2798/instrumentation/sidekiq/lib/opentelemetry/instrumentation/sidekiq/middlewares/server/tracer_middleware.rb)" instrumentation that deal with the same concern of propagating incoming context from external process, setting it into current process's context, and starting a span.

Reply:
Thanks, I agree in_span is a better choice for this instrumentation. Updated!